Makefile

```diff
@@ -0,0 +1,78 @@
+# vim: noexpandtab ts=8
+my_p=$(shell pwd -P)
+export GOPATH=$(shell echo $${GOPATH:-$$HOME/go})
+
+.PHONY: clean test verify update
+# Remove build artifaces
+# Example:
+#    make cllean
+#
+clean:
+	@rm -rf $(my_p)/_output
+
+# Build machine configs. Intended to be called via another target.
+# Example:
+#    make _machine-config-operator
+_machine-config-%:
+	WHAT=machine-config-$* $(my_p)/hack/build-go.sh
+
+# Build image for a component. Intended to be called via another target.
+# Example:
+#    make _image-machine-config-daemon
+_image-%:
+	@which podman 2> /dev/null &>1 || { echo "podman must be installed to build an image";  exit 1; }
+	WHAT=$* $(my_p)/hack/build-image.sh
+
+# Run unit tests
+# Example:
+#    make test
+test:
+	cd $(my_p) && go test -v ./...
+
+# Run the code generation tasks.
+# Example:
+#    make update
+update:
+	$(my_p)/hack/update-codegen.sh
+
+# Run verification steps
+# Example:
+#    make verify
+verify: test machine-configs
+	$(my_p)/hack/verify-style.sh
+	$(my_p)/hack/verify-codegen.sh
+
+define target_template =
+ .PHONY: machine-config-$(1)
+ machine-config-$(1): _machine-config-$(1)
+
+ .PHONY: image-machine-config-$(1)
+ image-machine-config-$(1): _image-machine-config-$(1)
```

Comment:
Nice use of templates!
We can add the dependency here, right? E.g.

```
  image-machine-config-$(1): _image-machine-config-$(1) _machine-config-$(1)
```

Reply:
Done, take another look. I reworked it a bit to make it slightly better. Specifically, the Makefile is more dynamic in the targets: any new `Dockerfiles.X` or new `cmd/X` that gets dropped in get Makefile targets automatically.